app/assets/javascripts/components/AccountMenu/ConfirmPassword.tsx

```diff
@@ -0,0 +1,225 @@
+      setEnableCustomServer(!enableCustomServer);
+    };
+
+    const handleSyncServerChange = (e: Event) => {
+      if (e.target instanceof HTMLInputElement) {
+        setSyncServer(e.target.value);
+        application.setCustomHost(e.target.value);
+      }
+    };
+
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === 'Enter') {
+        handleConfirmFormSubmit(e);
+      }
+    };
+
+    const handleConfirmFormSubmit = (e: Event) => {
+      e.preventDefault();
+
+      if (!password || password.length === 0) {
+        passwordInputRef?.current.focus();
+        return;
+      }
+
+      if (password === confirmPassword) {
+        setIsRegistering(true);
+        application
+          .register(email, password, isEphemeral, shouldMergeLocal)
+          .then((res) => {
+            if (res.error) {
+              throw new Error(res.error.message);
+            }
+            appState.accountMenu.closeAccountMenu();
+            appState.accountMenu.setCurrentPane(AccountMenuPane.GeneralMenu);
+          })
+          .catch((err) => {
+            console.error(err);
+            application.alertService.alert(err).finally(() => {
+              setPassword('');
+              handleGoBack();
+            });
+          })
+          .finally(() => {
+            setIsRegistering(false);
+          });
+      } else {
+        application.alertService
+          .alert(STRING_NON_MATCHING_PASSWORDS)
+          .finally(() => {
+            setConfirmPassword('');
+            passwordInputRef?.current.focus();
+          });
+      }
+    };
+
+    const handleGoBack = () => {
+      setMenuPane(AccountMenuPane.Register);
+    };
+
+    return (
+      <>
+        <div className="flex items-center px-3 mt-1 mb-3">
+          <IconButton
+            icon="arrow-left"
+            title="Go back"
+            className="flex mr-2 color-neutral"
+            onClick={handleGoBack}
+            focusable={true}
+            disabled={isRegistering}
+          />
+          <div className="sn-account-menu-headline">Confirm password</div>
+        </div>
+        <div className="px-3 mb-3 text-sm">
+          Because your notes are encrypted using your password,{' '}
+          <span className="color-dark-red">
+            Standard Notes does not have a password reset option
+          </span>
+          . If you forget your password, you will permanently lose access to
+          your data.
+        </div>
+        <form onSubmit={handleConfirmFormSubmit} className="px-3 mb-1">
+          <InputWithIcon
+            className="mb-2"
+            icon="password"
+            inputType={showPassword ? 'text' : 'password'}
+            placeholder="Confirm password"
+            value={confirmPassword}
+            onChange={handlePasswordChange}
+            onKeyDown={handleKeyDown}
+            toggle={{
+              toggleOnIcon: 'eye',
+              toggleOffIcon: 'eye',
```

Comment:
I think the icon here should switch between the regular eye icon to indicate a "Show" action and the icon with the eye crossed off to indicate a "Hide" action